streams/src/main/java/org/apache/kafka/streams/processor/internals/tasks/DefaultTaskExecutor.java

```diff
@@ -87,11 +88,29 @@ private void runOnce(final long nowMs) {
             if (currentTask == null) {
                 currentTask = taskManager.assignNextTask(DefaultTaskExecutor.this);
             } else {
```

Comment:
Why can we not process a newly assigned task immediately but we wait until the next iteration?

Reply:
We should be able to loop around very quickly here, and we'll have to duplicate the `currentTask != null` check... but okay if you prefer.